components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -438,12 +442,20 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
         //hang on to user profile before it's overwritten for analytics below
         const oldProfile = User.getProfile(user);
 
-        const allowedFields: (keyof User)[] = ["avatarUrl", "fullName", "additionalData"];
+        const allowedFields: (keyof User)[] = ["avatarUrl", "fullName"];
         for (const p of allowedFields) {
             if (p in partialUser) {
                 (user[p] as any) = partialUser[p];
             }
         }
+        // special treatment of nested additionalData
+        if (partialUser.additionalData) {
```

Comment:
Preferably, let's not use a field dedicated for user preferences/settings for something that actually controls the wheels or breaks. That check seems to work fine, I've tried to break without success, but I'd prefer not to have a this guarded places in a field which otherwise is free to change for users. 

There is an option that we can use to add field to tables without the downside of locking tables, we just need to add `NOT NULL DEFAULT '', ALGORITHM=INPLACE, LOCK=NONE` for the alter statement.

As we used this alternative the last time(s) discussed this matter, I'd prefer to stick to one pattern 🙏🏻

Reply:
Which discussion do you refer to?

Reply:
To avoid implementing the checks for nested attributes in the `additionalData` bucket we've moved the attribution override to `User` in https://github.com/gitpod-io/gitpod/pull/11522. The validation necessary is to disallow users setting the attribution target to a team id they are not a member of.